core/src/main/java/org/infinispan/container/offheap/OffHeapConcurrentMap.java

```diff
@@ -635,7 +635,7 @@ public boolean replace(WrappedBytes key, InternalCacheEntry<WrappedBytes, Wrappe
          @Override
          public Iterator<WrappedBytes> iterator() {
             // TODO: should we add a keyStream ?
-            return entryStream().map(InternalCacheEntry::getKey).iterator();
+            return entryStreamIncludingExpired().map(InternalCacheEntry::getKey).iterator();
```

Comment:
My thought was to extract our own interface that only has the things the data container implementations actually use, and expose `entryStreamIncludingExpiration()` directly instead of `keySet()` and the rest. But throwing an exception is also fine with me.